app_history.d.ts

```diff
@@ -59,7 +59,7 @@ interface AppHistoryEntryEventMap {
 declare class AppHistoryEntry extends EventTarget {
   readonly key: string;
   readonly id: string;
-  readonly url: string;
+  readonly url: string|null;
```

Comment:
The obvious thing in the code leaves the string empty rather than null. Catching that and converting it to null is presumably a 1-liner, but not the default.